src/query/query_executor.c

```diff
@@ -1160,6 +1160,21 @@ qexec_end_one_iteration (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE *
 	  GOTO_EXIT_ON_ERROR;
 	}
 
+      if (XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY))
+	{
+	  if (xasl->outptr_list)
+	    {
+	      if (xasl->outptr_list->valptrp)
+		{
+		  REGU_VARIABLE_LIST valptrp;
+		  for (valptrp = xasl->outptr_list->valptrp; valptrp; valptrp = valptrp->next)
+		    {
+		      valptrp->value.flags |= REGU_VARIABLE_UPD_INS_LIST;
+		    }
+		}
+	    }
```

Comment:
how about extracting it into a function? The same code exists in the changes.

Reply:
I modified it as a common routine.